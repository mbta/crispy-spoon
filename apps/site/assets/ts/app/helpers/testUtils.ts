/* eslint-disable */
import toJson, { Json } from "enzyme-to-json";
import { ReactWrapper } from "enzyme";
import {
  IndexedRouteStop,
  StopTree
} from "../../schedule/components/__schedule";

export const testRouteStopListFromStopTree = (
  tree: StopTree
): IndexedRouteStop[] =>
  Object.values(tree.byId).map((node, index) => ({
    ...node.value,
    routeIndex: index
  }));

export const createReactRoot = (): void => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
};

export const enzymeToJsonWithoutProps = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false,
    map: info => {
      if (!info.props) {
        return info;
      }

      // @ts-ignore is a <style></style> declaration
      if (info.node.type === "style") {
        return info;
      }
      // @ts-ignore is an element in a <noscript></noscript> declaration
      if (info.node.nodeType == undefined) {
        // @ts-ignore
        return info;
      }
      //@ts-ignore is a React component (not HTML element)
      if (info.node.nodeType.toString().includes("function")) {
        // @ts-ignore
        info.props = undefined;
      }

      return info;
    }
  });

export const toJsonEnzyme = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false
  });
