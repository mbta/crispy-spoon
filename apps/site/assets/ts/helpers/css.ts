import {
  isABusRoute,
  isACommuterRailRoute,
  isAGreenLineRoute,
  isASilverLineRoute
} from "../models/route";
import { Route } from "../__v3api";

export const routeToCSSClass = (route: string): string =>
  route.toLowerCase().replace(/[ _]/g, "-");

export const busClass = (route: Route): string =>
  isABusRoute(route) && !isASilverLineRoute(route) ? "bus-route-sign" : "";

export const routeToModeName = (route: Route): string => {
  if (isACommuterRailRoute(route)) return "commuter-rail";
  if (route.type === 4) return "ferry";
  if (route.id === "Red" || route.id === "Mattapan") return "red-line";
  if (route.id === "Orange") return "orange-line";
  if (route.id === "Blue") return "blue-line";
  if (isAGreenLineRoute(route)) return "green-line";
  if (isASilverLineRoute(route)) return "silver-line";
  if (isABusRoute(route)) return "bus";
  return "unknown";
};

export const routeBgClass = (route: Route): string =>
  `u-bg--${routeToModeName(route)}`;
