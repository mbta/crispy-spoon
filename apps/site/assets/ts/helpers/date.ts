// Returns a date in the form of "August 7, 2019"
export const formattedDate = (unformatted: string): string => {
  const [year, month, day] = unformatted
    .split(/-/)
    .map(part => Number.parseInt(part, 10));

  // Remember that months in JS are 0-indexed for some reason, hence "month - 1".
  const parsedDate = new Date(year, month - 1, day);
  return parsedDate.toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC"
  });
};

// Returns a date in the form of "Aug 7"
export const shortDate = (date: Date): string =>
  date.toLocaleDateString("en-US", {
    month: "short",
    day: "numeric",
    timeZone: "UTC"
  });

export default formattedDate;
