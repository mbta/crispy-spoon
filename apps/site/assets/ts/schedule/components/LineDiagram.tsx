import React, { ReactElement } from "react";
import { LineDiagramStop, StopData } from "./__schedule";
import {
  alertIcon,
  modeIcon,
  accessibleIcon,
  parkingIcon
} from "../../helpers/icon";
import { Alert, Route } from "../../__v3api";

interface Props {
  lineDiagram: LineDiagramStop[];
}

const maybeAlert = (alerts: Alert[]): ReactElement<HTMLElement> | null => {
  const highPriorityAlerts = alerts.filter(alert => alert.priority === "low");
  if (highPriorityAlerts.length === 0) return null;
  return (
    <>
      {alertIcon("c-svg__icon-alerts-triangle")}
      <span className="sr-only">Service alert or delay</span>
      &nbsp;
    </>
  );
};

const maybeTerminus = (stopData: StopData[]): StopData | undefined =>
  stopData.find((stop: StopData) => stop.type === "terminus");

const LineDiagram = ({
  lineDiagram
}: Props): ReactElement<HTMLElement> | null => (
  <>
    {lineDiagram.map(
      ({
        route_stop: routeStop,
        stop_data: stopData,
        alerts: stopAlerts
      }: LineDiagramStop) => (
        <div key={routeStop.id} className="m-schedule-line-diagram__stop">
          <div className="m-schedule-line-diagram__card-left">
            <div className="m-schedule-line-diagram__stop-name">
              {maybeAlert(stopAlerts)}
              {routeStop.name}
            </div>
            <div>
              {maybeTerminus(stopData) && maybeTerminus(stopData)!.branch}
            </div>
            <div className="m-schedule-line-diagram__connections">
              {routeStop.connections.map((route: Route) =>
                route.type === 3 && !route.name.startsWith("SL") ? (
                  <span
                    key={route.id}
                    className="c-icon__bus-pill m-schedule-line-diagram__connection u-bg--bus"
                  >
                    {route.name}
                  </span>
                ) : (
                  <span key={route.id}>{modeIcon(route.id)}</span>
                )
              )}
            </div>
          </div>
          <div>
            <div className="m-schedule-line-diagram__features">
              {routeStop.stop_features.includes("parking_lot")
                ? parkingIcon("c-svg__icon-parking-default")
                : null}
              {routeStop.stop_features.includes("access")
                ? accessibleIcon("c-svg__icon-acessible-default")
                : null}
              {routeStop.zone && (
                <span className="c-icon__cr-zone">{`Zone ${
                  routeStop.zone
                }`}</span>
              )}
            </div>
            <div>
              {stopData.map((stop: StopData) => (
                <div>
                  {stop.branch ? `${stop.branch}, ` : ""}
                  {stop.type}
                </div>
              ))}
            </div>
          </div>
        </div>
      )
    )}
  </>
);

export default LineDiagram;
