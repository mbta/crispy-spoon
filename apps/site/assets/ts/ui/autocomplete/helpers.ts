import { omit } from "lodash";
import { OnSubmitParams } from "@algolia/autocomplete-core";
import { OnStateChangeProps } from "@algolia/autocomplete-js";
import {
  ContentItem,
  Item,
  LocationItem,
  PopularItem,
  RouteItem,
  StopItem
} from "./__autocomplete";
import { isLGDown } from "../../helpers/media-breakpoints";

export function isStopItem(x: Item): x is StopItem {
  return Object.keys(x).includes("stop");
}

export function isRouteItem(x: Item): x is RouteItem {
  return Object.keys(x).includes("route");
}

export function isContentItem(x: Item): x is ContentItem {
  return Object.keys(x).includes("_content_type");
}

export const getTitleAttribute = (item: Item): string[] => {
  if (isStopItem(item)) {
    return ["stop", "name"];
  }
  if (isRouteItem(item)) {
    return ["route", "name"];
  }

  // @ts-ignore
  // eslint-disable-next-line no-underscore-dangle
  return Object.keys(item._highlightResult);
};

const navStateChange: (props: OnStateChangeProps<Item>) => void = ({
  state
}) => {
  // grey out the page and disable scrolling when search is open
  if (isLGDown()) {
    if (state.isOpen) {
      document.documentElement.dataset.navOpen = "true";
    } else {
      delete document.documentElement.dataset.navOpen;
    }
  }
};

export const STATE_CHANGE_HANDLERS: Record<
  string,
  (props: OnStateChangeProps<Item>) => void
> = {
  nav: navStateChange
};

export const SUBMIT_HANDLERS: Record<
  string,
  (props: OnSubmitParams<Item>) => void
> = {
  to_search_page({ state }) {
    window.Turbolinks.visit(`/search?query=${state.query}`);
  }
};

export type WithUrls<T> = T & { urls: Record<string, string> };

// The backend returns all possible URLs, use urlType to get the desired one
export const itemWithUrl = (
  initialItem: WithUrls<LocationItem | PopularItem>,
  urlType: string
): Omit<typeof initialItem, "urls"> => {
  const item = omit(initialItem, "urls");
  item.url = initialItem.urls[urlType];
  return item;
};
