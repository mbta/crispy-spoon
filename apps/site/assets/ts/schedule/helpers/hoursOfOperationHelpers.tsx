import React, { ReactElement } from "react";
import renderFa from "../../helpers/render-fa";
import { SchedulePDF } from "../components/__schedule";

const pdfLink = (pdf: SchedulePDF | null): ReactElement<HTMLElement> | null => {
  if (!pdf) {
    return null;
  }
  return (
    <div className="fs-16 pt-16">
      <a
        key={pdf.url}
        href={pdf.url}
        rel="noopener noreferrer"
        className="c-call-to-action"
        target="_blank"
      >
        {`Open full schedule and map PDF`}{" "}
        {renderFa("fa-arrow-up-right-from-square", "")}
      </a>
    </div>
  );
};

export default pdfLink;
