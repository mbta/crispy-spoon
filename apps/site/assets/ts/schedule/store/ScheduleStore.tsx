import { Reducer, createStore, Store } from "redux";
import { StoreAction, SelectedOrigin } from "../components/__schedule";
import { DirectionId } from "../../__v3api";
import { Mode as ModalMode } from "../components/schedule-finder/ScheduleFinderModal";

export interface StoreProps {
  selectedOrigin: SelectedOrigin | "";
  modalOpen: boolean;
  modalMode: ModalMode;
}

interface ActionData {
  selectedDirection?: DirectionId;
  selectedOrigin?: SelectedOrigin | "";
  modalOpen?: boolean;
  modalMode?: ModalMode;
}

export interface Action {
  type: StoreAction;
  newStoreValues: ActionData;
}

export const scheduleStoreReducer: Reducer<StoreProps, Action> = (
  state: StoreProps | undefined,
  action: Action
): StoreProps => {
  const newState: StoreProps = Object.assign(
    {
      selectedDirection: undefined,
      selectedOrigin: undefined,
      modalOpen: false,
      modalMode: "schedule"
    },
    state
  );

  switch (action.type) {
    case "INITIALIZE":
      return { ...newState, ...action.newStoreValues };
    case "CHANGE_ORIGIN":
      return {
        ...newState,
        selectedOrigin: action.newStoreValues.selectedOrigin!
      };
    case "OPEN_MODAL":
      return {
        ...newState,
        modalMode: action.newStoreValues.modalMode!,
        modalOpen: true
      };
    case "CLOSE_MODAL":
      return {
        ...newState,
        modalOpen: false
      };
    default:
      return newState;
  }
};

export const createScheduleStore = (): Store =>
  // create store with preloadedState:
  createStore(scheduleStoreReducer, {
    selectedOrigin: "",
    modalOpen: false,
    modalMode: "schedule"
  });

// initial 'dummy' value (will be overwritten when schedule-loader loads)
export const store: Store = createScheduleStore();

export const getCurrentState = (): StoreProps => store.getState();

export const storeHandler = (newValues: Action): void => {
  store.dispatch(newValues);
};

export const mapStateToProps = (state: StoreProps): StoreProps => ({
  ...state
});

export default mapStateToProps;
