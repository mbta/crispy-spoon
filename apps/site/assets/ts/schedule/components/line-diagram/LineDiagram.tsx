import React, { ReactElement, useState } from "react";
import { Provider } from "react-redux";
import { updateInLocation } from "use-query-params";
import SearchBox from "../../../components/SearchBox";
import { stopForId, stopIds } from "../../../helpers/stop-tree";
import useRealtime from "../../../hooks/useRealtime";
import { isSubwayRoute } from "../../../models/route";
import { Alert, DirectionId, Route } from "../../../__v3api";
import { getCurrentState, storeHandler } from "../../store/ScheduleStore";
import { changeOrigin } from "../ScheduleLoader";
import {
  IndexedRouteStop,
  RouteStop,
  SelectedOrigin,
  StopTree
} from "../__schedule";
import { createStopTreeCoordStore } from "./graphics/useTreeStopPositions";
import LineDiagramWithStops from "./LineDiagramWithStops";
import StopCard from "./StopCard";
import { alertsByStop } from "../../../models/alert";

interface Props {
  stopTree: StopTree | null;
  route: Route;
  routeStopList: IndexedRouteStop[];
  directionId: DirectionId;
  alerts: Alert[];
}

const stationsOrStops = (routeType: number): string =>
  [0, 1, 2].includes(routeType) ? "Stations" : "Stops";

const updateURL = (origin: SelectedOrigin, direction?: DirectionId): void => {
  /* istanbul ignore else */
  if (window) {
    // eslint-disable-next-line camelcase
    const newQuery = {
      "schedule_finder[direction_id]":
        direction !== undefined ? direction.toString() : "",
      "schedule_finder[origin]": origin
    };
    const newLoc = updateInLocation(newQuery, window.location);
    // newLoc is not a true Location, so toString doesn't work
    window.history.replaceState({}, "", `${newLoc.pathname}${newLoc.search}`);
  }
};

const LineDiagram = ({
  stopTree,
  routeStopList,
  route,
  directionId,
  alerts
}: Props): ReactElement<HTMLElement> => {
  const stopTreeCoordStore = stopTree
    ? createStopTreeCoordStore(stopTree)
    : createStopTreeCoordStore(routeStopList);
  const liveData = useRealtime(route, directionId, true);
  const [query, setQuery] = useState("");

  const allStops: RouteStop[] = stopTree
    ? stopIds(stopTree).map(stopId => stopForId(stopTree, stopId))
    : routeStopList;
  const filteredStops: RouteStop[] = allStops.filter(stop =>
    stop.name.toLowerCase().includes(query.toLowerCase())
  );

  const handleStopClick = (stop: RouteStop): void => {
    changeOrigin(stop.id);

    const currentState = getCurrentState();
    const { modalOpen: modalIsOpen } = currentState;

    updateURL(stop.id, directionId);

    if (currentState.selectedOrigin !== undefined && !modalIsOpen) {
      storeHandler({
        type: "OPEN_MODAL",
        newStoreValues: {
          modalMode: "schedule"
        }
      });
    }
  };

  return (
    <>
      {!isSubwayRoute(route) && (
        <h3 className="m-schedule-diagram__heading">
          {stationsOrStops(route.type)}
        </h3>
      )}
      <SearchBox
        id="stop-search"
        labelText={`Search for a ${stationsOrStops(route.type)
          .toLowerCase()
          .slice(0, -1)}`}
        onChange={setQuery}
        className="m-schedule-diagram__filter"
      />
      {query !== "" ? (
        <ol className="m-schedule-diagram m-schedule-diagram--searched">
          {filteredStops.length ? (
            filteredStops.map((stop: RouteStop) => (
              <StopCard
                key={stop.id}
                stopTree={stopTree}
                stopId={stop.id}
                routeStopList={routeStopList}
                alerts={alertsByStop(alerts, stop.id)}
                onClick={handleStopClick}
                liveData={liveData?.[stop.id]}
                searchQuery={query}
              />
            ))
          ) : (
            <div className="c-alert-item c-alert-item--low c-alert-item__top-text-container">
              No stops {route.direction_names[directionId]} to{" "}
              {route.direction_destinations[directionId]} matching{" "}
              <b className="u-highlight">{query}</b>. Try changing your
              direction or adjusting your search.
            </div>
          )}
        </ol>
      ) : (
        <Provider store={stopTreeCoordStore}>
          <LineDiagramWithStops
            stopTree={stopTree}
            routeStopList={routeStopList}
            route={route}
            directionId={directionId}
            alerts={alerts}
            handleStopClick={handleStopClick}
            liveData={liveData}
          />
        </Provider>
      )}
    </>
  );
};

export default LineDiagram;
