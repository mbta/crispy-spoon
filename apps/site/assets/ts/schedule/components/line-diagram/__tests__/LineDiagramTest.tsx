import React from "react";
import { mount, ReactWrapper } from "enzyme";
import * as swr from "swr";
import * as UseQueryParams from "use-query-params";
import LineDiagram from "../LineDiagram";
import { Route, RouteType } from "../../../../__v3api";
import { RouteStop, StopTree } from "../../__schedule";
import SearchBox from "../../../../components/SearchBox";
import * as ScheduleStore from "../../../store/ScheduleStore";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import * as store from "../../../store/ScheduleStore";
import userEvent from "@testing-library/user-event";
import { testRouteStopListFromStopTree } from "../../../../app/helpers/testUtils";

const stopTree: StopTree = {
  byId: {
    a: {
      id: "a",
      value: ({
        id: "a",
        name: "a",
        connections: [],
        stop_features: []
      } as unknown) as RouteStop
    },
    b: {
      id: "b",
      value: ({
        id: "b",
        name: "b",
        connections: [],
        stop_features: []
      } as unknown) as RouteStop
    },
    c: {
      id: "c",
      value: ({
        id: "c",
        name: "c",
        connections: [],
        stop_features: []
      } as unknown) as RouteStop
    }
  },
  edges: {
    a: { next: ["b"], previous: [] },
    b: { next: ["c"], previous: ["a"] },
    c: { next: [], previous: ["b"] }
  },
  startingNodes: ["a"]
};
const testRouteStopList = testRouteStopListFromStopTree(stopTree);
const route = {
  type: 3 as RouteType,
  name: "route 1",
  long_name: "route 1 long name",
  color: "F00B42",
  id: "route-1",
  direction_names: {
    0: "Outbound",
    1: "Inbound"
  },
  direction_destinations: {
    0: "Begin",
    1: "End"
  },
  description: "key_bus_route",
  "custom_route?": false,
  header: "",
  alerts: [],
  line_id: null
};

const useSWRSpy = jest.spyOn(swr, "default");
const storeHandlerSpy = jest.spyOn(ScheduleStore, "storeHandler");
const updateInLocationSpy = jest.spyOn(UseQueryParams, "updateInLocation");

describe("LineDiagram", () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(
      <LineDiagram
        stopTree={stopTree}
        routeStopList={testRouteStopList}
        route={route}
        directionId={1}
        alerts={[]}
      />
    );
  });

  afterEach(() => {
    wrapper.unmount();
    jest.clearAllMocks();
  });

  it("renders and matches snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("can filter stops by name", () => {
    const filter = wrapper.find(".m-schedule-diagram__filter").at(0);
    expect(filter.exists()).toBeTruthy();
    expect(filter.type()).toEqual(SearchBox);
  });

  it("should name stops or stations", () => {
    const subwayRoute = { ...route, type: 2 } as Route;
    const subwayWrapper = mount(
      <LineDiagram
        stopTree={stopTree}
        routeStopList={testRouteStopList}
        route={subwayRoute}
        directionId={1}
        alerts={[]}
      />
    );

    expect(wrapper.find(".m-schedule-diagram__heading").text()).toContain(
      "Stops"
    );
    expect(subwayWrapper.find(".m-schedule-diagram__heading").text()).toContain(
      "Stations"
    );
  });

  it("requests live data for most route types", () => {
    expect(useSWRSpy).toHaveBeenCalled();
    expect(useSWRSpy).toHaveBeenCalledWith(
      "/schedules/line_api/realtime?id=route-1&direction_id=1",
      expect.any(Function),
      expect.objectContaining({ refreshInterval: expect.any(Number) })
    );
  });

  it("should update the URL when the schedule finder modal is opened", () => {
    wrapper
      .find(".m-schedule-diagram__footer > button")
      .first()
      .simulate("click");

    expect(storeHandlerSpy).toHaveBeenCalledWith(
      expect.objectContaining({ type: "OPEN_MODAL" })
    );
    expect(updateInLocationSpy).toHaveBeenCalled();
  });

  it("should display the No Results card when a user doesn't query a stop", async () => {
    render(
      <LineDiagram
        stopTree={stopTree}
        routeStopList={testRouteStopList}
        route={route}
        directionId={0}
        alerts={[]}
      />
    );
    const search = screen.getByLabelText(/Search for a */);
    fireEvent.change(search, { target: { value: "Test Query Text" } });
    await waitFor(() => {
      expect(screen.getByText("Test Query Text")).toBeDefined();
      expect(screen.getByText(/No stops.*/)).toBeDefined();
    });
  });

  it("should display the Stop Card for each stop a user queries", () => {
    render(
      <LineDiagram
        stopTree={stopTree}
        routeStopList={testRouteStopList}
        route={route}
        directionId={0}
        alerts={[]}
      />
    );
    const search = screen.getByLabelText(/Search for a */);
    fireEvent.change(search, { target: { value: "a" } });

    // a is the stop name
    expect(screen.getByText("a")).toBeDefined();
  });

  it("should fire the open modal event when a user clicks on the results stop card", async () => {
    const storeHandlerSpy = jest.spyOn(store, "storeHandler");
    render(
      <LineDiagram
        stopTree={stopTree}
        routeStopList={testRouteStopList}
        route={route}
        directionId={0}
        alerts={[]}
      />
    );
    const search = screen.getByLabelText(/Search for a */);
    fireEvent.change(search, { target: { value: "a" } });

    const scheduleButton = await screen.findByText("View schedule");

    await userEvent.click(scheduleButton);

    await waitFor(() => {
      expect(storeHandlerSpy).toHaveBeenCalledWith({
        type: "OPEN_MODAL",
        newStoreValues: { modalMode: "schedule" }
      });
    });
  });
});
