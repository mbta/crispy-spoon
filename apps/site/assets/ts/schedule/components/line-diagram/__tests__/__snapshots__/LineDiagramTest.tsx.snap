// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineDiagram renders and matches snapshot 1`] = `
"<LineDiagram stopTree={{...}} routeStopList={{...}} route={{...}} directionId={1} alerts={{...}}>
  <h3 className=\\"m-schedule-diagram__heading\\">
    Stops
  </h3>
  <SearchBox id=\\"stop-search\\" labelText=\\"Search for a stop\\" onChange={[Function: bound dispatchAction]} className=\\"m-schedule-diagram__filter\\">
    <div className=\\"m-schedule-diagram__filter c-form__input-container\\">
      <label htmlFor=\\"stop-search\\" className=\\"sr-only\\">
        Search for a stop
      </label>
      <input className=\\"c-form__input\\" id=\\"stop-search\\" type=\\"text\\" onChange={[Function: onChange]} value=\\"\\" placeholder=\\"Search for a stop\\" />
      <button className=\\"c-form__submit-btn\\" aria-label=\\"search\\" type=\\"button\\" onClick={[Function: onClick]}>
        <span className=\\"notranslate c-svg__icon-search-reverse-default\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
      </button>
    </div>
  </SearchBox>
  <Provider store={{...}}>
    <LineDiagramWithStops stopTree={{...}} routeStopList={{...}} route={{...}} directionId={1} alerts={{...}} handleStopClick={[Function: handleStopClick]} liveData={[undefined]}>
      <div className=\\"m-schedule-diagram u-no-crowding-data\\">
        <Diagram stopTree={{...}} route={{...}} directionId={1} alerts={{...}} liveData={[undefined]}>
          <LiveVehicleIconSet isStart={true} stop={{...}} liveData={[undefined]} />
          <LiveVehicleIconSet isStart={false} stop={{...}} liveData={[undefined]} />
          <LiveVehicleIconSet isStart={false} stop={{...}} liveData={[undefined]} />
          <svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" aria-labelledby=\\"diagram-title diagram-desc\\" className=\\"line-diagram-svg bus\\" width=\\"36px\\" height=\\"100%\\" style={{...}}>
            <title id=\\"diagram-title\\">
              Line diagram for 
              route 1
            </title>
            <desc id=\\"diagram-desc\\">
              3 stops going Inbound to End
            </desc>
            <defs>
              <pattern id=\\"diagonalHatch\\" width=\\"10\\" height=\\"10\\" patternTransform=\\"rotate(45 0 0)\\" patternUnits=\\"userSpaceOnUse\\">
                <line className=\\"line-diagram-svg__line\\" y2={8} strokeWidth=\\"16px\\" />
              </pattern>
            </defs>
            <Line stopTree={{...}} fromId=\\"a\\" toId=\\"b\\" alerts={{...}}>
              <line className=\\"line-diagram-svg__line\\" strokeWidth=\\"10px\\" x1=\\"11px\\" x2=\\"11px\\" y1=\\"0px\\" y2=\\"0px\\" />
            </Line>
            <Line stopTree={{...}} fromId=\\"b\\" toId=\\"c\\" alerts={{...}}>
              <line className=\\"line-diagram-svg__line\\" strokeWidth=\\"10px\\" x1=\\"11px\\" x2=\\"11px\\" y1=\\"0px\\" y2=\\"0px\\" />
            </Line>
            <Stop stopId=\\"a\\">
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" data-shuttle={false} />
            </Stop>
            <Stop stopId=\\"b\\">
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" data-shuttle={false} />
            </Stop>
            <Stop stopId=\\"c\\">
              <circle className=\\"line-diagram-svg__stop\\" r=\\"4px\\" cx=\\"11px\\" cy=\\"0px\\" data-shuttle={false} />
            </Stop>
          </svg>
        </Diagram>
        <ol>
          <NextStopOrBranch stopTree={{...}} stopId=\\"a\\" alerts={{...}} mergingInBranches={{...}} splittingOffBranches={{...}} handleStopClick={[Function: handleStopClick]} liveData={[undefined]}>
            <StopCard stopTree={{...}} routeStopList={{...}} stopId=\\"a\\" alerts={{...}} onClick={[Function: handleStopClick]} liveData={[undefined]}>
              <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
                <section className=\\"m-schedule-diagram__content\\">
                  <GlxOpen pageType=\\"line-diagram\\" stopId=\\"a\\" />
                  <header className=\\"m-schedule-diagram__stop-heading\\">
                    <h4 className=\\"m-schedule-diagram__stop-link notranslate\\">
                      <a href=\\"/stops/a\\">
                        <MatchHighlight text=\\"a\\" matchQuery={[undefined]}>
                          <span>
                            a
                          </span>
                        </MatchHighlight>
                      </a>
                    </h4>
                    <div className=\\"m-schedule-diagram__features\\" />
                  </header>
                  <div className=\\"m-schedule-diagram__stop-details\\">
                    <div className=\\"m-schedule-diagram__connections\\" />
                  </div>
                  <footer className=\\"m-schedule-diagram__footer\\">
                    <button className=\\"btn btn-link\\" type=\\"button\\" onClick={[Function: onClick]}>
                      View schedule
                    </button>
                  </footer>
                </section>
              </li>
            </StopCard>
            <NextStopOrBranch stopTree={{...}} stopId=\\"b\\" alerts={{...}} mergingInBranches={{...}} splittingOffBranches={{...}} handleStopClick={[Function: handleStopClick]} liveData={[undefined]}>
              <StopCard stopTree={{...}} routeStopList={{...}} stopId=\\"b\\" alerts={{...}} onClick={[Function: handleStopClick]} liveData={[undefined]}>
                <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
                  <section className=\\"m-schedule-diagram__content\\">
                    <GlxOpen pageType=\\"line-diagram\\" stopId=\\"b\\" />
                    <header className=\\"m-schedule-diagram__stop-heading\\">
                      <h4 className=\\"m-schedule-diagram__stop-link notranslate\\">
                        <a href=\\"/stops/b\\">
                          <MatchHighlight text=\\"b\\" matchQuery={[undefined]}>
                            <span>
                              b
                            </span>
                          </MatchHighlight>
                        </a>
                      </h4>
                      <div className=\\"m-schedule-diagram__features\\" />
                    </header>
                    <div className=\\"m-schedule-diagram__stop-details\\">
                      <div className=\\"m-schedule-diagram__connections\\" />
                    </div>
                    <footer className=\\"m-schedule-diagram__footer\\">
                      <button className=\\"btn btn-link\\" type=\\"button\\" onClick={[Function: onClick]}>
                        View schedule
                      </button>
                    </footer>
                  </section>
                </li>
              </StopCard>
              <NextStopOrBranch stopTree={{...}} stopId=\\"c\\" alerts={{...}} mergingInBranches={{...}} splittingOffBranches={{...}} handleStopClick={[Function: handleStopClick]} liveData={[undefined]}>
                <StopCard stopTree={{...}} routeStopList={{...}} stopId=\\"c\\" alerts={{...}} onClick={[Function: handleStopClick]} liveData={[undefined]}>
                  <li className=\\"m-schedule-diagram__stop\\" style={{...}}>
                    <section className=\\"m-schedule-diagram__content\\">
                      <GlxOpen pageType=\\"line-diagram\\" stopId=\\"c\\" />
                      <header className=\\"m-schedule-diagram__stop-heading\\">
                        <h4 className=\\"m-schedule-diagram__stop-link notranslate\\">
                          <a href=\\"/stops/c\\">
                            <MatchHighlight text=\\"c\\" matchQuery={[undefined]}>
                              <span>
                                c
                              </span>
                            </MatchHighlight>
                          </a>
                        </h4>
                        <div className=\\"m-schedule-diagram__features\\" />
                      </header>
                      <div className=\\"m-schedule-diagram__stop-details\\">
                        <div className=\\"m-schedule-diagram__connections\\" />
                      </div>
                    </section>
                  </li>
                </StopCard>
              </NextStopOrBranch>
            </NextStopOrBranch>
          </NextStopOrBranch>
        </ol>
      </div>
    </LineDiagramWithStops>
  </Provider>
</LineDiagram>"
`;
